addons/website_sale_loyalty/tests/test_shop_sale_coupon.py

```diff
@@ -586,3 +586,59 @@ def test_confirm_points_as_public_user(self):
         order.action_quotation_send()
         order.with_context(access_token=order.access_token, user=public_user).action_confirm()
         self.assertEqual(loyalty_card.points, 1)
+
+    def test_archiving_loyalty_card_unlinks_draft_points_from_sale_order(self):
+        """
+        When a loyalty card has points accrued from a draft sale order, archiving the
+        card should unlink those draft points so they are no longer claimable on that order
+        """
+        website = self.env.ref('website.default_website')
+        product = self.env['product.product'].create({
+            'name': 'Product A',
+            'list_price': 10,
+            'sale_ok': True,
+        })
+        loyalty_program = self.env['loyalty.program'].create({
+            'name': 'Loyalty Program',
+            'program_type': 'loyalty',
+            'trigger': 'auto',
+            'applies_on': 'both',
+            'rule_ids': [
+                Command.create({
+                    'reward_point_mode': 'unit',
+                    'reward_point_amount': 10,
+                    'product_ids': [product.id],
+                }),
+            ],
+            'reward_ids': [
+                Command.create({
+                    'reward_type': 'discount',
+                    'discount': 50,
+                    'discount_mode': 'percent',
+                    'discount_applicability': 'order',
+                    'required_points': 10,
+                }),
+            ],
+        })
+        loyalty_card = self.env['loyalty.card'].create({
+            'program_id': loyalty_program.id,
+            'partner_id': self.steve.id,
+            'points': 0,
+        })
+        sale_order = self.empty_order
+        sale_order.write({
+            'website_id': website.id,
+            'order_line': [
+                Command.create({
+                    'product_id': product.id,
+                }),
+            ]
+        })
+        sale_order._update_programs_and_rewards()
+        claimable_rewards = sale_order._get_claimable_rewards()
+        self.assertTrue(claimable_rewards[loyalty_card])
+        loyalty_card.action_archive()
+        claimable_rewards = sale_order._get_claimable_rewards()
+        self.assertTrue(
+            loyalty_card not in claimable_rewards or not claimable_rewards[loyalty_card]
+        )
```

Comment:
```suggestion
        self.assertFalse(claimable_rewards.get(loyalty_card))
```
Doesn't this achieve the same result?